recipe-client-addon/lib/StudyStorage.jsm

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "IndexedDB", "resource://gre/modules/IndexedDB.jsm");
+
+this.EXPORTED_SYMBOLS = ["StudyStorage"];
+
+const DB_OPTIONS = {
+  namespace: "shield-study-addons",
+  version: 1,
+  storage: "persistent",
+  schema: {
+    name: { autoIncrement: false, unique: false },
+    version: { autoIncrement: false, unique: false },
+    description: { autoIncrement: false, unique: false },
+    installDate: { autoIncrement: false, unique: false },
+    studyEndDate: { autoIncrement: false, unique: false },
+    addonId: { autoIncrement: false, unique: true },
```

Comment:
We might run an experiment with the same add-on multiple times, which would lead to duplicate `addonId`s. I think `name` is a better candidate for the unique key, as we enforce unique experiment names on the server.